src/omnicore/omnicore.cpp

```diff
@@ -3048,6 +2888,54 @@ int mastercore::ClassAgnosticWalletTXBuilder(const std::string& senderAddress, c
     }
 }
 
+void CMPTxList::LoadActivations(int blockHeight)
+{
+    if (!pdb) return;
+
+    Slice skey, svalue;
+    Iterator* it = NewIterator();
+
+    PrintToLog("Loading feature activations from levelDB\n");
+
+    for(it->SeekToFirst(); it->Valid(); it->Next()) {
+       skey = it->key();
+       svalue = it->value();
+       string itData = svalue.ToString();
+
+       std::vector<std::string> vstr;
+       boost::split(vstr, itData, boost::is_any_of(":"), token_compress_on);
+       if (4 != vstr.size()) continue;
+       if (atoi(vstr[2]) != OMNICORE_MESSAGE_TYPE_ACTIVATION || atoi(vstr[0]) != 1) continue; // we only care about valid activations
+
+       uint256 hash(skey.ToString());
+       uint256 blockHash = 0;
+       CTransaction wtx;
+       CMPTransaction mp_obj;
+
+       if (!GetTransaction(hash, wtx, blockHash, true)) {
+           PrintToLog("ERROR: While loading activation transaction %s: tx in levelDB but does not exist.\n", hash.GetHex());
+           continue;
+       }
+       if (0 != ParseTransaction(wtx, blockHeight, 0, mp_obj)) {
```

Comment:
Uh oh! Consider (just in theory): the activation transaction confirmed in block 370000, and it was a class B transaction.

When `LoadActivations` is called at some point later, say with `nWaterlineBlock = 380000`, when class C was already activated, then it would not longer be parsed as class B, but class C.